Battleship/Ship.cs

```diff
@@ -1,12 +1,49 @@
+using System.Collections.Generic;
+
 namespace Battleship
 {
     public class Ship
     {
         private ShipClass _shipClass;
 
-        public Ship(ShipClass shipClass){
+        public Ship(ShipClass shipClass, Direction dir, Coordinate startingLocation){
             _shipClass = shipClass;
             ShipDamage = new bool[ShipClassSize.ShipSize[shipClass]];
+            Position.Coordinates.Add(startingLocation);
+            Coordinate modifiedLocation = startingLocation;
+            if(dir == Direction.North){
```

Comment:
I think all of this was why I didn't want to construct ships via rays (just a set of coordinates).  This logic should really be in the "ship picker" stuff I'm working on (or related).  But it's hack day and I'm just having fun learning what the Blazor buzz is all about.